### PR TITLE
Log out using App::log_out()

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

But only do we log out locally, we also need to log out on the server.

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: tests/js/session-tests.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ module.exports = {
                 TestCase.assertEqual(session.config.url, config.sync.url);
                 TestCase.assertEqual(session.config.user.identity, config.sync.user.identity);
                 TestCase.assertEqual(session.state, 'active');
-                user.logOut();
+                return user.logOut();
```
Comment (kraenhansen) [inline tests/js/session-tests.js:175]: Would it be too big of an unrelated change to make this function an `async function` and awaiting the `user` and then awaiting the `user.logOut()` afterwards? That will reduce the nesting caused by `.then` and the code calling this won't get the result of `logOut` returned (which it shouldn't need).

File: tests/js/encryption-tests.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ module.exports = {
                     partitionValue: "LoLo"
                 }
             });
-            user.logOut(); // FIXME: clearTestState() doesn't clean up enough and Realm.Sync.User.current might not work
+            return user.logOut(); // FIXME: clearTestState() doesn't clean up enough and Realm.Sync.User.current might not work
```
Comment (kraenhansen) [inline tests/js/encryption-tests.js:92]: Same comment as in the `session-tests.js`.

File: tests/js/session-tests.js (hunk 1 above)
Comment (kneth) [reply tests/js/session-tests.js:175]: It will require some work since we are spawning a new process (at least, it's more than a 10 minutes hack :smile:)

Comment (kraenhansen) [reply tests/js/session-tests.js:175]: What do you mean by "since we are spawning a new process"?
